pkg/app/extract.go

```diff
@@ -0,0 +1,158 @@
+package app
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/opcr-io/policy/oci"
+	perr "github.com/opcr-io/policy/pkg/errors"
+)
+
+// ExtractPolicyBundle extracts a policy bundle from OCI store to the specified directory
+func (c *PolicyApp) ExtractPolicyBundle(ociClient *oci.Oci, ref string, destDir string) error {
+	// Get reference descriptor
+	refDescriptor, err := c.getRefDescriptor(ociClient, ref)
+	if err != nil {
+		return perr.ErrExtractFailed.WithError(err).WithMessage("failed to get reference descriptor")
+	}
+
+	// Fetch the tarball from OCI store
+	reader, err := ociClient.GetStore().Fetch(c.Context, *refDescriptor)
+	if err != nil {
+		return perr.ErrExtractFailed.WithError(err).WithMessage("failed to fetch policy bundle")
+	}
+	defer func() {
+		if closeErr := reader.Close(); closeErr != nil {
+			c.UI.Problem().WithErr(closeErr).Msg("Failed to close OCI policy reader")
+		}
+	}()
+
+	// Create gzip reader
+	gzReader, err := gzip.NewReader(reader)
+	if err != nil {
+		return perr.ErrExtractFailed.WithError(err).WithMessage("failed to create gzip reader")
+	}
+	defer func() {
+		if closeErr := gzReader.Close(); closeErr != nil {
+			c.UI.Problem().WithErr(closeErr).Msg("Failed to close gzip reader")
+		}
+	}()
+
+	// Create destination directory if it doesn't exist
+	if err := os.MkdirAll(destDir, 0755); err != nil {
```

Comment:
The `os.MkdirAll` call is not needed when we enforce existance of `--untardir`

If we still need this, please use the existing filemode constant `x.OwnerReadWriteExecute`